pkg/endpoint/providers/kube/client.go

```diff
@@ -140,7 +140,8 @@ func (c Client) GetServiceForServiceAccount(svcAccount service.K8sServiceAccount
 				k8sServices, err := c.getServicesByLabels(selectorLabel, appNamspace)
 				if err != nil {
 					log.Error().Err(err).Msgf("Error retrieving service with label %v in namespace %s", selectorLabel, appNamspace)
-					return service.MeshService{}, errDidNotFindServiceForServiceAccount
+
+					return nil, errDidNotFindServiceForServiceAccount
```

Comment:
Are we planning to return nil rather than empty slices going forward ?

Reply:
it does save an allocation. I don't feel strongly either way